Inconsistent handling of user input when parsing flags

Example(handled by the flag library, so the panic in config.go never happens):
```
kolide serve --osquery_node_key_size=abcd
Error running root command: invalid argument "abcd" for --osquery_node_key_size=abcd: strconv.ParseInt: parsing "abcd": invalid syntax
``` 

But sometimes panics happen, and they're ugly. 
```
kolide serve -c /foo/bar/baz                                                                                                                                         ⏎
Using config file:  /foo/bar/baz
panic: Error reading config: open /foo/bar/baz: no such file or directory

goroutine 1 [running]:
github.com/kolide/kolide/server/config.Manager.loadConfigFile(0xc42011e0f0, 0xc42013a240, 0xc42013d260)
	/Users/victor/go/src/github.com/kolide/kolide/server/config/config.go:400 +0x28b
github.com/kolide/kolide/server/config.Manager.LoadConfig(0xc42011e0f0, 0xc42013a240, 0xc42013d260, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/victor/go/src/github.com/kolide/kolide/server/config/config.go:177 +0x7f
github.com/kolide/kolide/cli.createServeCmd.func1(0xc42013b200, 0xc42014a760, 0x0, 0x2)
	/Users/victor/go/src/github.com/kolide/kolide/cli/serve.go:48 +0xee
github.com/kolide/kolide/vendor/github.com/spf13/cobra.(*Command).execute(0xc42013b200, 0xc42014a600, 0x2, 0x2, 0xc42013b200, 0xc42014a600)
	/Users/victor/go/src/github.com/kolide/kolide/vendor/github.com/spf13/cobra/command.go:603 +0x22b
github.com/kolide/kolide/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42013a240, 0xc420117f50, 0xc420140dd0, 0xc420140d8a)
	/Users/victor/go/src/github.com/kolide/kolide/vendor/github.com/spf13/cobra/command.go:689 +0x339
github.com/kolide/kolide/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42013a240, 0xc420117f48, 0x1)
	/Users/victor/go/src/github.com/kolide/kolide/vendor/github.com/spf13/cobra/command.go:648 +0x2b
github.com/kolide/kolide/cli.Launch()
	/Users/victor/go/src/github.com/kolide/kolide/cli/root.go:22 +0x1d1
main.main()
	/Users/victor/go/src/github.com/kolide/kolide/main.go:16 +0x20
```

--- a/server/config/config.go
+++ b/server/config/config.go
@@ -2,6 +2,7 @@ package config
 
 import (
 	"fmt"
+	"os"
 	"strings"
 	"time"
 
@@ -486,11 +487,12 @@ func (man Manager) loadConfigFile() {
 	man.viper.SetConfigFile(configFile)
 	err := man.viper.ReadInConfig()
 
-	fmt.Println("Using config file: ", man.viper.ConfigFileUsed())
-
 	if err != nil {
-		panic("Error reading config: " + err.Error())
+		fmt.Println("Error loading config file:", err)
+		os.Exit(1)
 	}
+
+	fmt.Println("Using config file: ", man.viper.ConfigFileUsed())
 }
 
 // TestConfig returns a barebones configuration suitable for use in tests.

